src/main/java/com/example/Restaurant_Management_System_REST_API/service/ReportService.java

```diff
@@ -0,0 +1,388 @@
+package com.example.Restaurant_Management_System_REST_API.service;
+
+import com.example.Restaurant_Management_System_REST_API.DTO.MenuRecordDTOs.MenuRecordForOrderDTO;
+import com.example.Restaurant_Management_System_REST_API.DTO.RestaurantOrderMenuRecordDTO.RestaurantOrderMenuRecordDTO;
+import com.example.Restaurant_Management_System_REST_API.DTO.TableDTO.TableForReportDTO;
+import com.example.Restaurant_Management_System_REST_API.exception.NotFoundInDatabaseException;
+import com.example.Restaurant_Management_System_REST_API.model.Category;
+import com.example.Restaurant_Management_System_REST_API.model.ContactDetails;
+import com.example.Restaurant_Management_System_REST_API.model.OrderStatus;
+import com.example.Restaurant_Management_System_REST_API.model.entity.*;
+import com.example.Restaurant_Management_System_REST_API.repository.*;
+import lombok.AllArgsConstructor;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+import org.modelmapper.ModelMapper;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.stereotype.Service;
+import org.springframework.web.servlet.mvc.method.annotation.StreamingResponseBody;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Service
+@AllArgsConstructor
+public class ReportService {
+
+    private final InventoryItemRepository inventoryItemRepository;
+    private final CustomerRepository customerRepository;
+    private final MenuRecordRepository menuRecordRepository;
+    private final ReservationRepository reservationRepository;
+    private final RestaurantOrderRepository restaurantOrderRepository;
+    private final SupplierRepository supplierRepository;
+    private final TableRepository tableRepository;
+    private final ModelMapper modelMapper;
+    private final RestaurantOrderMenuRecordRepository restaurantOrderMenuRecordRepository;
```

Comment:
To avoid using these repositories directly you could use services that use those. This could have some benefits, but it's okay in this form, too :)